Index: src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- src/chrome/browser/ui/gtk/browser_window_gtk.cc	(revision 235101)
+++ src/chrome/browser/ui/gtk/browser_window_gtk.cc	(working copy)
@@ -2416,7 +2416,11 @@
 }
 
 // static
+#if 0 // berkelium patch: remove
 BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+#else // berkelium patch: added
+BrowserWindow* BerkeliumCreateBrowserWindow(Browser* browser) {
+#endif // berkelium patch: end
   BrowserWindowGtk* browser_window_gtk = new BrowserWindowGtk(browser);
   browser_window_gtk->Init();
   return browser_window_gtk;
Index: src/chrome/app/chrome_main.cc
===================================================================
--- src/chrome/app/chrome_main.cc	(revision 235101)
+++ src/chrome/app/chrome_main.cc	(working copy)
@@ -39,3 +39,13 @@
   return content::ContentMain(argc, argv, &chrome_main_delegate);
 #endif
 }
+
+#if 1 // berkelium patch: added
+#include "chrome/browser/ui/browser_window.h"
+
+BrowserWindow* BerkeliumCreateBrowserWindow(Browser*);
+
+BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+  return BerkeliumCreateBrowserWindow(browser);
+}
+#endif // berkelium patch: end
Index: src/chrome/chrome_exe.gypi
===================================================================
--- src/chrome/chrome_exe.gypi	(revision 235101)
+++ src/chrome/chrome_exe.gypi	(working copy)
@@ -30,6 +30,9 @@
         'app/metro_driver_win.h',
         '../content/app/startup_helper_win.cc',
       ],
+      'dependencies': [
+            '../skia/skia.gyp:skia',
+      ],
       'mac_bundle_resources': [
         'app/app-Info.plist',
       ],
