Index: src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- src/chrome/browser/ui/gtk/browser_window_gtk.cc	(revision 235101)
+++ src/chrome/browser/ui/gtk/browser_window_gtk.cc	(working copy)
@@ -2415,8 +2415,16 @@
           wm_type == ui::WM_XFWM4);
 }
 
+#if 1 // berkelium patch: added
+BrowserWindow* (*BerkeliumHook_CreateBrowserWindow)(Browser*) = NULL;
+#endif // berkelium patch: end
+
 // static
 BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+#if 1 // berkelium patch: added
+  if(BerkeliumHook_CreateBrowserWindow != NULL)
+    return BerkeliumHook_CreateBrowserWindow(browser);
+#endif // berkelium patch: end
   BrowserWindowGtk* browser_window_gtk = new BrowserWindowGtk(browser);
   browser_window_gtk->Init();
   return browser_window_gtk;
Index: src/chrome/browser/ui/views/frame/browser_view.cc
===================================================================
--- src/chrome/browser/ui/views/frame/browser_view.cc	(revision 235101)
+++ src/chrome/browser/ui/views/frame/browser_view.cc	(working copy)
@@ -2521,8 +2521,16 @@
 #endif
 }
 
+#if 1 // berkelium patch: added
+BrowserWindow* (*BerkeliumHook_CreateBrowserWindow)(Browser*) = NULL;
+#endif // berkelium patch: end
+
 // static
 BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+#if 1 // berkelium patch: added
+  if(BerkeliumHook_CreateBrowserWindow != NULL)
+    return BerkeliumHook_CreateBrowserWindow(browser);
+#endif // berkelium patch: end
   // Create the view and the frame. The frame will attach itself via the view
   // so we don't need to do anything with the pointer.
   BrowserView* view = new BrowserView();
Index: src/content/browser/web_contents/web_contents_view_gtk.cc
===================================================================
--- src/content/browser/web_contents/web_contents_view_gtk.cc	(revision 235101)
+++ src/content/browser/web_contents/web_contents_view_gtk.cc	(working copy)
@@ -76,10 +76,18 @@
 
 }  // namespace
 
+#if 1 // berkelium patch: added
+CONTENT_EXPORT WebContentsViewPort* (*BerkeliumHook_CreateWebContentsView)(WebContentsImpl*, WebContentsViewDelegate*, RenderViewHostDelegateView**);
+#endif // berkelium patch: end
+
 WebContentsViewPort* CreateWebContentsView(
     WebContentsImpl* web_contents,
     WebContentsViewDelegate* delegate,
     RenderViewHostDelegateView** render_view_host_delegate_view) {
+#if 1 // berkelium patch: added
+  if(BerkeliumHook_CreateWebContentsView != NULL)
+    return BerkeliumHook_CreateWebContentsView(web_contents, delegate, render_view_host_delegate_view);
+#endif // berkelium patch: end
   WebContentsViewGtk* rv = new WebContentsViewGtk(web_contents, delegate);
   *render_view_host_delegate_view = rv;
   return rv;
Index: src/content/browser/web_contents/web_contents_view_win.cc
===================================================================
--- src/content/browser/web_contents/web_contents_view_win.cc	(revision 235101)
+++ src/content/browser/web_contents/web_contents_view_win.cc	(working copy)
@@ -20,10 +20,18 @@
 #include "ui/gfx/screen.h"
 
 namespace content {
+#if 1 // berkelium patch: added
+CONTENT_EXPORT WebContentsViewPort* (*BerkeliumHook_CreateWebContentsView)(WebContentsImpl*, WebContentsViewDelegate*, RenderViewHostDelegateView**);
+#endif // berkelium patch: end
+
 WebContentsViewPort* CreateWebContentsView(
     WebContentsImpl* web_contents,
     WebContentsViewDelegate* delegate,
     RenderViewHostDelegateView** render_view_host_delegate_view) {
+#if 1 // berkelium patch: added
+  if(BerkeliumHook_CreateWebContentsView != NULL)
+    return BerkeliumHook_CreateWebContentsView(web_contents, delegate, render_view_host_delegate_view);
+#endif // berkelium patch: end
   WebContentsViewWin* rv = new WebContentsViewWin(web_contents, delegate);
   *render_view_host_delegate_view = rv;
   return rv;
Index: src/chrome/chrome.gyp
===================================================================
--- src/chrome/chrome.gyp	(revision 235101)
+++ src/chrome/chrome.gyp	(working copy)
@@ -112,6 +112,9 @@
   'conditions': [
     ['OS!="ios"', {
       'includes': [
+# BERKELIUM PATCH: added
+        '../../berkelium.gypi',
+# BERKELIUM PATCH: end
         'app/policy/policy_templates.gypi',
         'chrome_browser_extensions.gypi',
         'chrome_dll.gypi',
