Index: src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- src/chrome/browser/ui/gtk/browser_window_gtk.cc	(revision 202711)
+++ src/chrome/browser/ui/gtk/browser_window_gtk.cc	(working copy)
@@ -2376,7 +2376,11 @@
 }
 
 // static
+#if 0 // berkelium patch: remove
 BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+#else // berkelium patch: added
+BrowserWindow* BerkeliumCreateBrowserWindow(Browser* browser) {
+#endif // berkelium patch: end
   BrowserWindowGtk* browser_window_gtk = new BrowserWindowGtk(browser);
   browser_window_gtk->Init();
   return browser_window_gtk;
Index: src/chrome/app/chrome_main.cc
===================================================================
--- src/chrome/app/chrome_main.cc	(revision 202711)
+++ src/chrome/app/chrome_main.cc	(working copy)
@@ -32,3 +32,13 @@
   return content::ContentMain(argc, argv, &chrome_main_delegate);
 #endif
 }
+
+#if 1 // berkelium patch: added
+#include "chrome/browser/ui/browser_window.h"
+
+BrowserWindow* BerkeliumCreateBrowserWindow(Browser*);
+
+BrowserWindow* BrowserWindow::CreateBrowserWindow(Browser* browser) {
+  return BerkeliumCreateBrowserWindow(browser);
+}
+#endif // berkelium patch: end
